### DISCUSSION TITLE
it's annoying to see hundreds of `Expression has type "Any"` errors on every line after the line with the actual error

### DISCUSSION THREAD

**Discussion by DetachHead** [KotlinIsland/basedmypy#162] [Q&A] [answered]
ie. a function signature changes, so the result where it's called becomes `Any` because it's now being called wrong. so then every line past that point has a "type contains any" error

it would be cool if it only showed the error that's actually causing the issue instead of cluttering the output with hundreds of useless error messages

any solution to this issue would need to be mindful not to cause another instance of #152

NB: only happens with overloads

**Comment by KotlinIsland** [comment]
Current behavior:
```py
@overload
def foo(a: str) -> str: ...
@overload
def foo(a: int) -> int: ...

a = foo(None)  # error, inferred type of a is Any
print(a)  # Any? seems kinda sus
```

potential behavior:
```py
@overload
def foo(a: str) -> str: ...
@overload
def foo(a: int) -> int: ...

a = foo(None)  # error, inferred type of a is str
print(a)  # str, that's not sus :)
```

This is basically the way that TS behaves.

**Reply by DetachHead** [reply]
making the return type `Never` on an error might be a good solution, because in cases where it no longer knows what the return type could be there wouldn't be any follow up errors
```py
@overload
def foo(a: str) -> bool: ...
@overload
def foo(a: int) -> str: ...

a = foo([1])  # error, inferred type of a is Never
print(a == True)  # no error
```

**Reply by KotlinIsland** [reply]
You would then get a second error about an unreachable line.

**Reply by DetachHead** [reply]
a second error is nowhere near as bad as dozens more errors, but even then we could add logic to not show the unreachable error in this case

**Comment by KotlinIsland** [comment, marked as answer]
This issue describes the situation and solution: python/mypy/issues/12016
